src/relay/transforms/to_mixed_precision.cc

```diff
@@ -0,0 +1,409 @@
+                                                  const DataType& wanted_dtype) {
+    Array<Expr> new_args;
+    Array<Type> new_arg_types;
+    for (size_t i = 0; i < cur_args.size(); i++) {
+      Expr cur_arg = cur_args[i];
+      Type cur_arg_type = cur_arg_types[i];
+      Expr new_arg = CastArg(cur_arg, cur_arg_type, wanted_dtype);
+      Type new_arg_type = GetType(new_arg);
+      new_args.push_back(new_arg);
+      new_arg_types.push_back(new_arg_type);
+    }
+    return {new_args, new_arg_types};
+  }
+
+ public:
+  using MixedModeMutator::VisitExpr_;
+
+  explicit MixedPrecisionPass(DataType mixed_precision_type = DataType::Float(16),
+                              bool ignore_missing_ops = true, bool warn_missing_ops = true)
+      : MixedModeMutator(),
+        mixed_precision_type(mixed_precision_type),
+        ignore_missing_ops(ignore_missing_ops),
+        warn_missing_ops(warn_missing_ops) {
+    if (!mixed_precision_type.is_float() && !mixed_precision_type.is_bfloat16())
+      LOG(FATAL) << "Only support IEEE floating point mixed precision types and bfloat16 got "
+                 << mixed_precision_type;
+  }
+
+  Expr Rewrite_(const CallNode* pre_call_node, const Expr& post) final {
+    const CallNode* post_call_node = post.as<CallNode>();
+    if (!post_call_node) {
+      LOG(FATAL) << "Expected a CallNode for the rewrite got " << post;
+    }
+
+    Expr cur_op = post_call_node->op;
+
+    // Get info on the operation being called:
+    // conversion category (int), accumulation dtype (str), output dtype (str)
+    MixedTypeConversionCategory initial_category;
+    DataType accumulation_dtype, output_dtype;
+    if (cur_op.as<FunctionNode>()) {
+      // Avoid messing with functions to avoid changing signature
+      initial_category = MIXED_PRECISION_NEVER;
+      accumulation_dtype = DataType::Float(32);
+      output_dtype = DataType::Float(32);
+    } else if (cur_op.as<OpNode>()) {
+      static auto attr_map =
+          Op::GetAttrMap<FTVMMixedPrecisionConversionType>("FTVMMixedPrecisionConversionType");
+      Op op = Downcast<Op>(cur_op);
+      if (attr_map.count(op)) {
+        // Calculate the conversion category and dtypes from registered attribute.
+        FTVMMixedPrecisionConversionType func = attr_map[op];
+        Array<ObjectRef> op_descriptor =
+            func(GetRef<Call>(pre_call_node), DLDataType2String(mixed_precision_type));
+
+        int64_t op_conversion_type = Downcast<Integer>(op_descriptor[0])->value;
+        initial_category = static_cast<MixedTypeConversionCategory>(op_conversion_type);
+        accumulation_dtype = DataType(String2DLDataType(Downcast<String>(op_descriptor[1])));
+        output_dtype = DataType(String2DLDataType(Downcast<String>(op_descriptor[2])));
+      } else {
+        if (!ignore_missing_ops) LOG(FATAL) << "Op " << op->name << " not in conversion lists!";
+        if (warn_missing_ops) LOG(WARNING) << "Op " << op->name << " not in conversion lists!";
```

Comment:
Looks like we can merge these configs to be `verbose` or something like it.

Reply:
So you suggest a single flag `verbose` which just controls emission of warnings?

Reply:
Yeah but maybe don't call it verbose because `!ignore_mssing_ops` will throw errors and terminate the pass. Maybe the following?

```
allow_missing_ops: int
0: Does not allow any missing ops. Will throw errors and terminate the pass when encountering any. 
1: Allow missing ops but throw warnings.
2: Allow missing ops and silently ignore them.
```

Either 1 or 2 can be the default mode.
Meanwhile, whatever the behavior we decided at the end, it would be better to also control the message. For example, the current implementation has 2 problems:
1. The pass stops when it sees the first missing op.
2. If a model has **one** missing op but it appears 100 times, then users will see 100 warning messages.

As a result, it would be better to first collect unique missing ops with their appearances (e.g., `(op, times)`), and print out it at the end of this pass according to the user-provided flag/mode.

Reply:
Done, it's now a single flag `missing_op_mode`